include/substrait-mlir/Dialect/Substrait/IR/SubstraitOps.td

```diff
@@ -716,6 +717,109 @@ def Substrait_FilterOp : Substrait_RelOp<"filter", [
   }];
 }
 
+def Substrait_KeyComparisonOp : Substrait_ExpressionOp<"compare", [
+    DeclareOpInterfaceMethods<InferTypeOpInterface>
+  ]> {
+  let summary = "Key comparison expression";
+  let description = [{
+    Represents a comparison between two field references or expressions
+    used in join conditions.
+    
+    Example:
+    
+    ```mlir
+    %3 = field_reference %arg0[0] : tuple<si32, si32>
+    %4 = field_reference %arg1[0] : tuple<si32, si32, si32>
+    %5 = compare not_distinct_from %3, %4 : (si32, si32) -> si1
+    ```
+  }];
+  
+  let arguments = (ins
+    Substrait_ExpressionType:$lhs,
+    Substrait_ExpressionType:$rhs,
+    OptionalAttr<SimpleComparisonType>:$comparison_type,
+    OptionalAttr<UI32Attr>:$custom_function_id
```

Comment:
If `custom_function_id` is deliberately not implemented, i'd say remove this argument and write a TODO in the `description` field.